DCC.Core.Types.schema.json

```diff
@@ -7,7 +7,7 @@
     "dose_posint": {
       "description": "Dose Number / Total doses in Series: positive integer, range: [1,9]",
       "type": "integer",
-      "minimum": 1,
+      "minimum": 0,
```

Comment:
Yep fits with DICOM Type 2 - at least the approximation we can do here with JSON
However, a positive integer by defintion does not include zero e.g. https://mathworld.wolfram.com/PositiveInteger.html
So we would need to drop `posint` as a type designator here and call it something different

Reply:
I like this  PR :-)
I need to re-work some schema parts anyway so I'm going for a rename of `posint` to `nonnegative integer` (`non-neg`, sth like that) - and thus including zero, which helps us with the DICOM Type 2 approach, see also e.g. https://mathworld.wolfram.com/NonnegativeInteger.html

Reply:
I prefer this naming, less _systems hungarian_ 👍

Reply:
> Technicaly maybe, but logically, do we really want to alow non positive entries? I don't see the business need.

Business need: when you don't have total dose number (is an optional in e.g. HL7) but the EU Regs currently mandate a number must be supplied.
Is actually a fundamental generic issue for many data fields that may not be present in the original source data but are mandated bu EU regs to be present in the certificate - was discussed at length (again) in the Co-ordinated Actions meeting yesterday afternoon / evening.
And the whole point of this PR was the DICOM Type 2 approach, which directly addresses this business issue at an ISO global level for decades already, which is why I chose it as a valid and - for those in the field - well-known and ISO approved way of working

Reply:
So, we agreed to day that we should potentially change only the upper limit. Lower limit should remain 1.

Reply:
No, it was a proposal.
TBD Monday.
In any case:
 * the dose number should definitely be at least of numeric value 1 - by defintion, with the issuance of a vacc cert. 
 * the total doses may well simply not be present
  
Putting 1 for total dose means you may well have just changed the authorized medication as approved by EMA. It cannot be the case for example, that vaccine X has both 1 total dose and 2 total dose: the total dose is a function of the vaccine, not the person and will not change unless via regulatory (EMA) approval - but then it usually comes on the market as some variant vaccine clearly marked as being distinct in some way from the previous version

Reply:
Gaby, I disagree. Number of doeses in the series is a funtion of the vaccination protocol not of the vaccine itself. Foa any vaccin, you could have multiple vaccination protocols: 1 for healthy adults, another for youth and yet another for adults recovered from covid. In each protocol, you MUST know how many doses should be given.

Reply:
hmmm true to a point, yes different regulatory authorities could require different total doses, but that is not so usual, at least in my experience

> In each protocol, you MUST know how many doses should be given.

Yes, but the medical record data source for vacc has `total doses` marked as an optional field and we also see in practice that this data field is simply left empty / un-recorded (as is the case for so many "optional" fields).

Thus whilst you will have to know the total doses for the purposes of delivering medical care (the protocol), this does not necessarily translate to this same information being recorded in and available from a medical information system - primarily because people are often pressed for time and anything that is optional is (understandably) simply not filled-in.

If you have an empty `total dose` field you could try to fill this in data from supplemental information (from where? do we have an officially approved source we can recommend?). However, simply setting `total doses` to 1 or 2 to simply match the `doses given` field would be sometimes correct, sometimes not. I guess that was my point.

Ideally we would just drop this 1 / 2 , 2 / 2 thing altogether and just use the HL7 `status` field ("completed"). What we need for admin purposes in the end is "completed" (or "not completed") and if we could get that from an authoritative medical data source then that would be IMHO the least controversial option here. However, AFAIK the HSC has provided the requirement for `dose number` and `total doses` and may well have had some other requirements / constraints for specifying it like this.

Reply:
`total dose` from where?: one option could be to traverse the protocol if field at top level in vacc record is empty, sth like that

Reply:
> Ideally we would just drop this 1 / 2 , 2 / 2 thing altogether and just use the HL7 status field ("completed"). What we need for admin purposes in the end is "completed" (or "not completed") and if we could get that from an authoritative medical data source then that would be IMHO the least controversial option here. However, AFAIK the HSC has provided the requirement for dose number and total doses and may well have had some other requirements / constraints for specifying it like this.

Would be the best solution, however would also settle for 'non optimal' solution.